### PR TITLE
[device]: Fix a bug that psuutil cannot access gpio sysfs to get PSU status  

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- What I did**
Fix bug that found on  #1781

**- How I did it**
Update code to support dynamic sysfs gpio base.

**- How to verify it**
In SONiC run psuutil cli to test the following case:
PSU1: Absence
PSU1: Presence, no-power
PSU2: Absence
PSU1: Presence, no-power

### CODE REVIEW

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 1]
```diff
@@ -18,33 +21,43 @@ def __init__(self):
...
         export_file = "/sys/class/gpio/export"
-        direction_file = '/sys/class/gpio/gpio' + str(gpio_base+pinnum) + '/direction' 
-
         try:
             with open(export_file, 'w') as fd:
                 fd.write(str(gpio_base+pinnum))
         except Exception as error:
-            logging.error("Unable to export gpio ", pinnum)
+            logging.error("Unable to export gpio ", str(gpio_base+pinnum))
+            sys.exit(0)
```
Comment (lguohan) [inline device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: why exit 0 here? the plugin should not do exit

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 2]
```diff
@@ -54,6 +67,7 @@ def read_psu_statuses(self, pinnum):
                 retval = fd.read()
         except Exception as error:
             logging.error("Unable to open ", gpio_file, "file !")
+            sys.exit(0)
```
Comment (lguohan) [inline device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: the plugin should not do exit.

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 3]
```diff
@@ -18,33 +20,41 @@ def __init__(self):
...
         gpio_base = self.psu[0]['base']
         export_file = "/sys/class/gpio/export"
-        direction_file = '/sys/class/gpio/gpio' + str(gpio_base+pinnum) + '/direction' 
-
         try:
             with open(export_file, 'w') as fd:
                 fd.write(str(gpio_base+pinnum))
         except Exception as error:
-            logging.error("Unable to export gpio ", pinnum)
+            logging.error("Unable to export gpio ", str(gpio_base+pinnum))
```
Comment (jleveque) [inline device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: This is improper syntax. Please change to 

`logging.error("Unable to export gpio " + str(gpio_base+pinnum))`

Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: OK, I will change it.

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 4]
```diff
@@ -18,33 +20,41 @@ def __init__(self):
...
+        sys_gpio_dir = "/sys/class/gpio"
+        try:
+            for r in os.listdir(sys_gpio_dir):
+                if "gpiochip" in r:
+                    return int(r[8:],10)
+        except Exception as error:
+            logging.error("Unable to get gpio base")
+
+
     def init_psu_gpio(self, pinnum):
```
Comment (lguohan) [inline device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: this init_psu_gpio should move to your driver, we should not do init psu gpio on-the-fly.

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 5]
```diff
@@ -53,7 +63,7 @@ def read_psu_statuses(self, pinnum):
             with open(gpio_file, 'r') as fd:
                 retval = fd.read()
         except Exception as error:
```
Comment (lguohan) [inline device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: we should not catch general exception here, we should only catch exception we can handle in this case the io or file exception.

Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: OK, I will change it

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py (hunk 4 above)
Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: @pphuchar Please add this to our platform driver

File: device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py [hunk 6]
```diff
@@ -53,7 +63,7 @@ def read_psu_statuses(self, pinnum):
...
-            logging.error("Unable to open ", gpio_file, "file !")
+            logging.error("Unable to open " + gpio_file + "file !")
```
[the remark this answers is not in the document]
Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/plugins/psuutil.py]: OK , I will add raise statement to report the error

**Pull request merged**